code/game/machinery/magnet.dm

```diff
@@ -41,6 +41,10 @@
 	spawn()
 		magnetic_process()
 
+/obj/machinery/magnetic_module/Destroy()
+	SSradio.remove_object(src, freq)  // i have zero idea what the hell is going on
```

Comment:
The radio signal system is awful. So many devices communicate on this "virtual radio" system which involves posting a signal to a "channel", everything then receives it and then decides if its for that or not. 

Its a total mess and I want to kill it.